src/Core/Impl/IO/DirectoryInfoProxy.cs

```diff
@@ -40,6 +42,17 @@ public sealed class DirectoryInfoProxy : IDirectoryInfo {
                 .EnumerateFileSystemInfos()
                 .Select(CreateFileSystemInfoProxy);
 
+        public IEnumerable<IFileSystemInfo> EnumerateFileSystemInfos(string[] includeFiles, string[] excludeFiles) {
+            Matcher matcher = new Matcher();
+            matcher.AddIncludePatterns(includeFiles.IsNullOrEmpty() ? new[] { "**/*" } : includeFiles);
+            matcher.AddExcludePatterns(excludeFiles ?? Enumerable.Empty<string>());
+            var matchResult = matcher.Execute(new DirectoryInfoWrapper(_directoryInfo));
```

Comment:
I'd add a retry at least and definitely we don't want to crash LS just b/c some file is not accessible.